### PR TITLE
new filter: znc-adminlog

### PR DESCRIPTION
**Pull request by girst**
[ZNC](https://wiki.znc.in) is an IRC bouncer (used for keeping logs while you are away). I've written a filter that will ban users who failed to log in. It requires the adminlog module (comes with ZNC) to be loaded, as ZNC keeps silent about failed login attempts otherwise. 

I'm using it myself with this jail configuration:
```
[znc-adminlog]
enabled = true
filter = znc-adminlog
action = iptables[name=IRC, port=ircd, protocol=tcp]
logpath = /home/znc/.znc/moddata/adminlog/znc.log
maxretry = 5
```

### CODE REVIEW

**Comment by girst** [comment]
ping @sebres and others: do you want this? (I'm not upset if you don't, but please let me know either way)

File: config/filter.d/znc-adminlog.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Fail2Ban filter for ZNC (requires adminlog module)
+#
+# to use this module, enable the adminlog module from within ZNC and point
+# logpath to its logfile (e.g. /home/znc/.znc/moddata/adminlog/znc.log).
+
+[Definition]
+
+failregex = failed to login from <HOST>
```
Comment (sebres) [inline config/filter.d/znc-adminlog.conf]: Please use anchored regex here:
```diff
failregex = failed to login from <HOST>
failregex = ^\[\] \[[^]]+\] failed to login from <HOST>$
```

Comment (girst) [reply config/filter.d/znc-adminlog.conf]: thanks, implemented!

**Comment by girst** [comment]
Can this please be merged, @sebres and others, since it has been approved for over two months now?

I've forced-pushed one little improvement: the config now includes the [default znc service paths](https://wiki.znc.in/Running_ZNC_as_a_system_daemon)

**Comment by sebres** [comment]
> I've forced-pushed one little improvement ...

This is not recommended to do if previous revisions were approved... You could add this amend as new commit too.
Now I should check again the PR completely.

**Comment by girst** [comment]
Ping @sebres: This has been approved for over a month now. Can this please be merged? Thanks!

**Comment by sebres** [comment]
Indeed... Thx for the reminder.
Small notice by the way - if no real host names expected in log it is advisable to use tag `<ADDR>` instead of `<HOST>`.
You could update it or inform us whether it is expected (then I'll merge it as is).

**Comment by girst** [comment]
On Wed, Jun 12, 2019 at 02:53:30AM -0700, Sergey G. Brester wrote:
>Indeed... Thx for the reminder.
>Small notice by the way - if no real host names expected in log it is 
>advisable to use tag `<ADDR>` instead of `<HOST>`.
>You could update it or inform us whether it is expected (then I'll 
>merge it as is).

Thanks for the tip! ZNC doesn't appear to resolve hostnames. Do you want 
me to amend the commit or push it as a seperate one for you to squash 
later?

**Comment by sebres** [comment]
add it simply as new commit and push in this branch

**Comment by girst** [comment]
Done! let me know if anything else needs attention / changes ;-)

**Comment by sebres** [comment]
Merged in 809e7c4e82aa457f2e24d26b6c31dc5a61580357 in both major branches.

**Pull request closed without merging**